cypress/integration/unit_tests/color_function_util_test.js

```diff
@@ -0,0 +1,133 @@
+import * as addDisasterUtil from '../../../docs/import/add_disaster_util.js';
+import {populateColorFunctions, withColor} from '../../../docs/import/color_function_util.js';
+import {createTrs, setDisasterAndLayers} from '../../support/import_test_util.js';
+import {addFirebaseHooks, loadScriptsBeforeForUnitTests} from '../../support/script_loader.js';
+
+describe('Unit tests for add_disaster page', () => {
+  loadScriptsBeforeForUnitTests('ee', 'firebase', 'jquery');
+  addFirebaseHooks();
+
+  const property = 'color-function';
+  let colorFunctionEditor;
+  let writeToFirebaseStub;
+
+  before(() => {
+    cy.wrap(firebase.auth().signInWithCustomToken(firestoreCustomToken));
+
+    colorFunctionEditor =
+        $(document.createElement('div')).prop('id', 'color-fxn-editor').hide();
+    colorFunctionEditor.append(
+        makeTypeDiv('single'), makeTypeDiv('continuous'),
+        makeTypeDiv('discrete'));
+    $(document.body).append(colorFunctionEditor);
+
+    populateColorFunctions();
+
+    /**
+     * Makes one of the type divs (mimicking html in add_disaster.html)
+     * @param {string} id
+     * @return {JQuery<HTMLDivElement>}
+     */
+    function makeTypeDiv(id) {
+      return $(document.createElement('div'))
+          .prop('id', id)
+          .hide()
+          .addClass('color-type-div');
+    }
+  });
+
+  beforeEach(() => {
+    writeToFirebaseStub = cy.stub(addDisasterUtil, 'updateLayersInFirestore');
+  });
+
+  it('switches schemas and writes data', () => {
+    const layer = {
+      'color-function': {
+        'current-style': 2,
+        'columns': {
+          'wings': {'min': 0, 'max': 2, 'values': [0, 1, 2]},
+          'legs': {'min': 0, 'max': 100, 'values': [0, 2, 4, 8, 100]},
+        },
+        'color': 'yellow',
+      },
+    };
+    setDisasterAndLayers([layer]);
+
+    const td = withColor($(document.createElement('td')), layer, property);
+    const row = createTrs(1);
+    row[0].append(td);
+
+    expect(colorFunctionEditor.is(':visible')).to.be.false;
+
+    td.trigger('click');
+    expect(colorFunctionEditor.is(':visible')).to.be.true;
+    expect(writeToFirebaseStub).to.be.calledOnce;
```

Comment:
we write on color dialog display, even if there's no actual change? Seems like it would be good to avoid that, but it might be hard with my suggestion to set data unconditionally. Will comment below.

Reply:
Done - with assertion that nothing is called on display